addons/account/models/account_reconcile_model.py

```diff
@@ -667,33 +667,27 @@ def _get_invoice_matching_amls_domain(self, st_line, partner):
 
         return aml_domain
 
+    def _get_text_location_list(self):
+        fields = [
+            self.match_text_location_label and 'payment_ref',
+            self.match_text_location_note and 'narration',
+            self.match_text_location_reference and 'ref',
+        ]
+        return [f for f in fields if f]
+
     def _get_invoice_matching_st_line_tokens(self, st_line):
         """ Parse the textual information from the statement line passed as parameter
         in order to extract from it the meaningful information in order to perform the matching.
 
         :param st_line: A statement line.
         :return: A list of tokens, each one being a string.
         """
-        st_line_text_values = st_line._get_st_line_strings_for_matching(allowed_fields=(
-            'payment_ref' if self.match_text_location_label else None,
-            'narration' if self.match_text_location_note else None,
-            'ref' if self.match_text_location_reference else None,
-        ))
+        st_line_text_values = st_line._get_st_line_strings_for_matching(allowed_fields=self._get_text_location_list())
+
         significant_token_size = 4
         tokens = []
         for text_value in st_line_text_values:
-            for token in (text_value or '').split():
-                # The token is too short to be significant.
-                if len(token) < significant_token_size:
-                    continue
-
-                formatted_token = ''.join(x for x in token if x.isdecimal())
-
-                # The token is too short after formatting to be significant.
-                if len(formatted_token) < significant_token_size:
-                    continue
-
-                tokens.append(formatted_token)
+            tokens += [token for token in (text_value or '').split() if len(token) >= significant_token_size]
```

Comment:
You are removing the check on decimals here. You want to match INV/2023/00001 with "invoice 202300001" for example. Instead, keep the code as before but make an extra exact matching if `payment_ref` or `ref` are a single word. For example: `payment_ref="turlututu"` should match a line with `"turlututu"` as name